litex/build/generic_platform.py

```diff
@@ -132,6 +132,9 @@ def _resource_type(resource):
 class ConnectorManager:
     def __init__(self, connectors):
         self.connector_table = dict()
+        self.add_connectors(connectors)
+
+    def add_connectors(self, connectors):
```

Comment:
Maybe we should have a `add_connector` and move the `for connector in connectors` loop in __init__?

Reply:
I think testing `connector` type is more easy, avoid the overhead to call this method multiple time and allows to pass a serie of connectors: this method may be used with a scalar (tuple) or with a list (as done for **lite** variant for the **sipeed Tang Primer 20k** dock):

```
if isinstance(connector, tuple):
    connector = [connector]
```

Thanks to that this method will be more generic and avoid `[]` boilerplate when only one connector must be added.